File: leetcode/editor/cn/268_丢失的数字_test.py
import unittest
from typing import List


class TestSolution(unittest.TestCase):

    def setUp(self) -> None:
        self.solution = Solution()

    def test_givenNormal_when_thenSuccess(self):
        missing = self.solution.missingNumber([3, 0, 1])
        self.assertEqual(2, missing)


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        ans = 0
        for x in nums:
            ans ^= x
        for x in range(len(nums) + 1):
            ans ^= x
        return ans


# leetcode submit region end(Prohibit modification and deletion)


if __name__ == '__main__':
    unittest.main()

File: leetcode/editor/cn/268_丢失的数字_test.go
package leetcode

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func Test_givenNormal_when268_thenSuccess(t *testing.T) {
	assert.Equal(t, 2, missingNumber([]int{3, 0, 1}))
}

//leetcode submit region begin(Prohibit modification and deletion)
func missingNumber(nums []int) int {
	n := len(nums)
	sum := n * (n + 1) / 2
	arrSum := 0
	for _, val := range nums {
		arrSum += val
	}
	return sum - arrSum
}

//leetcode submit region end(Prohibit modification and deletion)
